core-backend/common/src/mock.rs

```diff
@@ -264,3 +267,94 @@ impl BackendExt for MockExt {
         Ok(())
     }
 }
+
+#[derive(Debug)]
+pub struct MockMemory {
+    pages: Vec<u8>,
+}
+
+impl MockMemory {
+    pub fn new(initial_pages: u32) -> Self {
+        let mut pages = Vec::with_capacity(initial_pages as usize * WASM_PAGE_SIZE);
+        unsafe {
+            pages.set_len(initial_pages as usize * WASM_PAGE_SIZE);
+        }
+        Self { pages }
+    }
+
+    fn page_index(&self, offset: u32) -> Option<usize> {
+        let index = offset as usize / WASM_PAGE_SIZE;
+        if (index as usize) < self.pages.len() / WASM_PAGE_SIZE {
+            Some(index as usize)
+        } else {
+            None
+        }
+    }
+}
+
+impl Memory for MockMemory {
+    type GrowError = ();
+
+    fn grow(&mut self, pages: WasmPage) -> Result<(), Self::GrowError> {
+        let new_size = self.pages.len() + (pages.raw() as usize) * WASM_PAGE_SIZE;
+        self.pages.reserve(new_size);
+        unsafe {
+            self.pages.set_len(new_size);
+        }
```

Comment:
Also, why not `resize`? I guess we can fill the mock memory with some value, aren't we?

Reply:
Done